internal/signal/signal.go

```diff
@@ -101,6 +101,10 @@ func Insert(ctx context.Context, dbc *sql.DB, workflow, run string, signalType i
 
 func completeChecksForever(ctx context.Context, cl replay.Client, dbc *sql.DB) {
```

Comment:
Rather change the signature to accept leader election, `getCtx func() context.Context`